src/main/cpp/native_c4query.cc

```diff
@@ -128,9 +128,9 @@ Java_com_couchbase_lite_internal_core_C4Query_createIndex(JNIEnv *env, jclass cl
     jstringSlice expressionsJSON(env, jexpressionsJSON);
     jstringSlice language(env, jlanguage);
     C4IndexOptions options = {};
-    slice sLang = language;
+    FLSlice sLang = language;
     if(sLang.buf != NULL)
-        options.language = (const char *)sLang.buf;
+        options.language = language.c_str();
```

Comment:
do we need the `if(sLang.buf != NULL)`? options.language == NULL

Reply:
I don't think we need it. I will remove it and test.